metakb/cli.py

```diff
@@ -0,0 +1,171 @@
+"""
+Provide CLI utility for performing data collection, transformation, and upload
+to graph datastore.
+"""
+import click
+from os import environ
+from metakb.database import Graph
+from metakb.harvesters import CIViC, MOAlmanac
+from metakb.transform import CIViCTransform, MOATransform
+import logging
+from metakb import PROJECT_ROOT
+from disease.database import Database as DiseaseDatabase
+from disease.schemas import SourceName as DiseaseSources
+from disease.cli import CLI as DiseaseCLI
+from therapy.database import Database as TherapyDatabase
+from therapy import ACCEPTED_SOURCES as TherapySources
+from therapy import SOURCES as TherapySourceLookup
+from therapy.cli import CLI as TherapyCLI
+from gene.database import Database as GeneDatabase
+from gene.cli import CLI as GeneCLI
+
+
+logger = logging.getLogger('metakb')
+logger.setLevel(logging.DEBUG)
+
+
+class CLI:
+    """Update database."""
+
+    @click.command()
+    @click.option(
+        '--db_url',
+        help=('URL endpoint for the application database. Can also be '
+              'provided via environment variable METAKB_DB_URL.')
+    )
+    @click.option(
+        '--db_username',
+        help='Username to provide to application database.'
+    )
+    @click.option(
+        '--db_password',
+        help='Password to provide to application database.'
+    )
+    @click.option(
+        '--check_normalizers',
+        is_flag=True,
+        help=('Check normalizer data repositories and exit if not '
+              'initialized.')
+    )
+    @click.option(
+        '--initialize',
+        is_flag=True,
+        help='Initialize all normalizer data repositories.'
+    )
+    @click.option(
+        '--normalizer_db_url',
+        help='URL endpoint of normalizer database.'
+    )
+    def update_metakb_db(db_url, db_username, db_password,
+                         check_normalizers=False, initialize=False,
+                         normalizer_db_url=''):
+        """Execute data harvest and transformation from resources and upload
+        to graph datastore.
+        """
+        uninitialized_srcs = []
+        if check_normalizers:
```

Comment:
We should do the `db_url` check in this if statement or before it.

Reply:
I'm also wondering if we should just include this in `--initialize` and for whichever normalizers don't have that data, we load it for them.